airbyte_cdk/sources/declarative/decoders/composite_raw_decoder.py

```diff
@@ -42,6 +45,31 @@ def parse(
             yield from self.inner_parser.parse(gzipobj)
 
 
+@dataclass
+class JsonParser(Parser):
+    encoding: str = "utf-8"
+
+    def parse(self, data: BufferedIOBase) -> Generator[MutableMapping[str, Any], None, None]:
+        raw_data = data.read()
+        try:
+            body_json = orjson.loads(raw_data.decode(self.encoding))
+        except Exception:
+            try:
+                body_json = json.loads(raw_data.decode(self.encoding))
+            except Exception as exc:
+                raise AirbyteTracedException(
```

Comment:
LGTM overall, but I guess we could get rid of nested try/except block and use something like this:

```suggestion
    body_json = self._parse_orjson(raw_data) or self._parse_json(raw_data)
    if body_json is None:
            raise AirbyteTracedException(
            ....


def _parse_orjson(self, raw_data: bytes):
    try:
        return orjson.loads(raw_data.decode(self.encoding))
    except orjson.JSONDecodeError:
         logger.error(...)
         return None

def _parse_json(self, raw_data: bytes):
    try:
        return json.loads(raw_data.decode(self.encoding))
    except json.JSONDecodeError:
        logger.warning(...)
        return None
```